dashboard/src/utils/filterDataset.js

```diff
@@ -0,0 +1,10 @@
+export const filterData = (dataArray, startDate, endDate, controllerValue) => {
+  let res = dataArray.filter((data) => {
+    return (
+      data.controller.includes(controllerValue) &&
+      new Date(data.metadata["dataset.created"].split(":")[0]) >= startDate &&
+      new Date(data.metadata["dataset.created"].split(":")[0]) <= new Date(endDate)
```

Comment:
You're going to fix things so that both `startDate` and `endDate` are `Date` objects, right?  😉